contracts/src/Teleporter/upgrades/TeleporterUpgradeable.sol

```diff
@@ -85,13 +103,47 @@ abstract contract TeleporterUpgradeable is ITeleporterReceiver {
         _setMinTeleporterVersion(version);
     }
 
+    /**
+     * @dev Pauses a Teleporter address from interacting with this contract.
+     * After pausing a Teleporter address, it will no longer be able to deliver messages
+     * to this contract, and this contract will not send messages through that Teleporter address.
+     * The address does not need to be registered with the Teleporter registry.
+     * Emits a {TeleporterAddressPaused} event if successfully paused.
+     * Requirements:
+     *
+     * - `msg.sender` must have Teleporter upgrade access.
+     * - `teleporterAddress` is not already paused.
+     */
+    function pauseTeleporterAddress(address teleporterAddress) public virtual {
+        _checkTeleporterUpgradeAccess();
```

Comment:
Nit: We should probably rename this to `_checkTeleporterAdminAccess` or something similar now that it's used for more than just upgrading.

Reply:
I prefer to leave as is, because some access controls might not have an "admin", for example in Warp off-chain message. Also, I think pausing still has relation to integration with the upgradeability.